src/ai/backend/manager/models/session.py

```diff
@@ -1082,6 +1082,68 @@ async def get_sgroup_managed_sessions(
         return result.scalars().all()
 
 
+async def finalize_running(
```

Comment:
Unlike `SessionRow.transit_session_status()` this hook does not have a safety gear to prevent session status from being updated to an illegal state. https://github.com/lablup/backend.ai/blob/bde8d203dd00223d962a3e11772d9a2319b770a9/src/ai/backend/manager/models/session.py#L762-L764 Is it okay to do so?